cou/steps/__init__.py

```diff
@@ -293,6 +293,16 @@ async def run(self) -> None:
         logger.debug("No coroutine to run for %s", repr(self))
 
 
+class AvailabilityZoneUpgradePlan(UpgradePlan):
```

Comment:
non-blocking: we are using `HypervisorGroup` class for type-agnostic grouping of the hypervisor machines, I think this should be updated to follow the same approach